Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -799,256 +771,202 @@ public final class ManifestLoader: ManifestLoaderProtocol {
-                                    // Append the runtime output to any compiler output we've received.
-                                    evaluationResult.compilerOutput = (evaluationResult.compilerOutput ?? "") + runOutput
-                                }
-
-                                // Return now if there was an error.
-                                if runResult.exitStatus != .terminated(code: 0) {
-                                    // TODO: should this simply be an error?
-                                    // return completion(.failure(ProcessResult.Error.nonZeroExit(runResult)))
-                                    evaluationResult.errorOutput = evaluationResult.compilerOutput
-                                    return completion(.success(evaluationResult))
-                                }
-
-                                // Read the JSON output that was emitted by libPackageDescription.
-                                let jsonOutput: String = try localFileSystem.readFileContents(jsonOutputFile)
-                                evaluationResult.manifestJSON = jsonOutput
-
-                                delegateQueue?.async {
-                                    delegate?.didEvaluate(
-                                        packageIdentity: packageIdentity,
-                                        packageLocation: packageLocation,
-                                        manifestPath: manifestPath,
-                                        duration: evaluationStart.distance(to: .now())
-                                    )
-                                }
-
-                                completion(.success(evaluationResult))
-                            } catch {
-                                completion(.failure(error))
-                            }
-                        }
-                    }
+
+                var environment = ProcessEnv.vars
+                #if os(Windows)
+                let windowsPathComponent = runtimePath.pathString.replacingOccurrences(of: "/", with: "\\")
+                environment["Path"] = "\(windowsPathComponent);\(environment["Path"] ?? "")"
+                #endif
+
+                let runResult = try await TSCBasic.Process.popen(arguments: cmd, environment: environment)
+                fclose(jsonOutputFileDesc)
+
+                if let runOutput = try (runResult.utf8Output() + runResult.utf8stderrOutput()).spm_chuzzle() {
+                    // Append the runtime output to any compiler output we've received.
+                    evaluationResult.compilerOutput = (evaluationResult.compilerOutput ?? "") + runOutput
                 }
-            } catch {
-                return callbackQueue.async {
-                    completion(.failure(error))
+
+                // Return now if there was an error.
+                if runResult.exitStatus != .terminated(code: 0) {
```

Comment:
Not sure if this is in scope for this change, but this looks like it wants to be a guard

Reply:
I was just following the previous structure 1:1, but I agree this is a good opportunity for a `guard` conversion.